src/components/NewTodo.tsx

```diff
@@ -2,20 +2,29 @@ import React, { useRef } from 'react';
 import './NewTodo.css';
 
 // what is better: type or interface? still not sure what the diff is, need to double check
+// I'd recommend checking this => https://github.com/microsoft/TypeScript/wiki/Performance, section: Preferring Interfaces over Intersections.
+// Following that, and considering our Props definition CAN be extended, I'd use export interface NewTodoProps { ... }
+
 // also, advice on how to manage the different TS interfaces - each in a separate file? they should have their own folder, I guess, but on what level of the project?
+// Declare interfaces of component props in the same file as the Functional Component (my personal preference).
 type NewTodoProps = {
+
     onAddTodo: (text: string) => void;
 }
 
-const NewTodo: React.FC<NewTodoProps> = ({ onAddTodo }) => {
+// Avoid using React.FC<T> https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/function_components/
+const NewTodo = ({ onAddTodo }: NewTodoProps) => {
     const textInputRef = useRef<HTMLInputElement>(null); //always use .current when working with refs
 
     const submitHandler = (event: React.FormEvent) => {
         event.preventDefault();
+        // maybe a better name is simply todo or todoText? 
         const enteredText = textInputRef.current!.value;
```

Comment:
I would also avoid using the non-null assertion. Much rather have a check before line 22:
If (!textInputRef.current) {
 return;
 }